app/support/end_reservation_only.rb

```diff
@@ -0,0 +1,34 @@
+class EndReservationOnly
+  def perform
+    order_details.each do |od|
+      od.transaction do
+        expire_reservation(od)
+      end
+    end
+  end
+
+  private
+
+  def order_details
+    reservation_only_order_details
+  end
+
+  def reservation_only_order_details
+    OrderDetail.purchased_active_reservations
+      .merge(Reservation.in_progress)
```

Comment:
Reservation only reservation will not have an actual_start_at, so I'm not sure this'll capture it.